programs/clearing_house/src/controller/position.rs

```diff
@@ -313,7 +321,52 @@ pub fn update_position_and_market(
     Ok(pnl)
 }
 
-pub fn update_position_with_base_asset_amount(
+pub fn update_position_and_market_with_fee(
+    position: &mut MarketPosition,
+    market: &mut Market,
+    delta: &PositionDelta,
+    fee_to_market: u128,
+) -> ClearingHouseResult<i128> {
+    // update user position
+    let pnl = update_position(Some(position), market, delta, true)?;
+
+    // update Market implicit position
+    let amm_pnl = update_position(
+        None,
+        market,
+        &PositionDelta {
+            base_asset_amount: -delta.base_asset_amount,
+            quote_asset_amount: delta.quote_asset_amount,
+        },
+        false,
+    )?;
+
+    market.amm.market_position.unsettled_pnl = market
+        .amm
+        .market_position
+        .unsettled_pnl
+        .checked_add(
+            amm_pnl
+                .checked_add(cast_to_i128(fee_to_market)?)
+                .ok_or_else(math_error!())?,
+        )
+        .ok_or_else(math_error!())?;
+
+    Ok(pnl)
+}
+
+pub fn update_position_and_market(
+    position: &mut MarketPosition,
+    market: &mut Market,
+    delta: &PositionDelta,
+) -> ClearingHouseResult<i128> {
+    // update user position
+    let pnl = update_position_and_market_with_fee(position, market, delta, 0)?;
```

Comment:
This function is weird, did you make it to avoid having to change the unit tests?

Reply:
updated